contracts/bridge/contracts/BasicInboundChannel.sol

```diff
@@ -47,39 +57,22 @@ contract BasicInboundChannel {
             ),
             "Channel: Invalid proof"
         );
-        verifyMessages(messages, beefyMMRLeaf, blockHeader);
+        verifyMessages(messages, beefyMMRLeaf);
         processMessages(messages);
     }
 
-    //TODO: verifyMessages should accept all needed proofs
-    function verifyMessages(Message[] memory messages, bytes memory beefyMMRLeaf, bytes memory blockHeader)
+    function verifyMessages(Message[] memory messages, BeefyMMRLeaf memory leaf)
         internal
         view
         returns (bool success)
     {
-        // struct BeefyMMRLeaf {
-        //     uint32 parentNumber;
-        //     bytes32 parentHash;
-        //     bytes32 parachainHeadsRoot;
-        //     uint64 nextAuthoritySetId;
-        //     uint32 nextAuthoritySetLen;
-        //     bytes32 nextAuthoritySetRoot;
```

Comment:
Do we need the nextAuthoritySetRoot in the mmr leaf now?

Reply:
Substrate put `nextAuthoritySetRoot` in the mmr leaf.
There is no need for solidity to know where it is. 
May be Darwinia could do the decision.

Reply:
OKm `nextAuthoritySetRoot` is in the payload of the commitment signed by grandpa validators, so it does not need to be in mmr.